public/app/core/utils/datemath.ts

```diff
@@ -14,26 +27,29 @@ export function parse(text, roundUp?, timezone?) {
     return moment(text);
   }
 
+  // Explicit cast to make TS happy
+  const stringTime = text as string;
```

Comment:
So everything from this line and to the end `text` is a string. Could we just do
```ts
if (typeof text !== "string") {
        return text;
}
```

and then remove the stringTime and the cast?

Reply:
That makes sense, I am not a fan of force casting either.